### ISSUE TITLE
Train with one nuclei type

### ISSUE THREAD

**Issue by Vadori** [TIO-IKIM/CellViT#50]
Hi!

Thanks for sharing. I am trying to train the model on a dataset with only one nuclei type. If I set the num_nuclei_classes =2, does the branch for nuclei type prediction get constructed? I will dig into the code, but it would be nice to get an answer/confirmation if you know this already :) Should I delete the branch and anything related to it manually, as suggested in another issue for getting predictions on a single tissue type?

Thank you!!

**Comment by FabianHoerst** [comment]
Hi!

I think the most straight way would be to remove everything related to the classification head (type prediction). The head is not removed automatically. You need to adapt the model by removing the branch:
https://github.com/TIO-IKIM/CellViT/blob/ebcc23c77c84b7d104c1b124ce9f6b71113ae434/models/segmentation/cell_segmentation/cellvit.py#L149

You also need to modify the following files:
`cell_segmentation/trainer/trainer_cellvit.py`and `cell_segmentation/experiments/experiment_cellvit_pannuke.py`

**Comment by Vadori** [comment]
Dear Fabian,

thank you for your response, much appreciated!! I actually modified the scripts to manage the one tissue type case and it runs perfectly, not sure I will have time to adapt the code to the one nuclei type case, but it appears the modifications would be similar. Again, thanks for sharing!!

**Comment by Vadori** [comment]
Hi again @FabianHoerst,

Training was running fine until epoch 25, when the encoder gets unfrozen. At this point the loss may become nan in certain epochs until I guess the model is somehow corrupted and the loss is nan all the times. Has this ever happened to you during training? It may be due to the mixed precision, so I am trying now to set mixed precision to False. Any other suggestions? Any response, especially if you faced this same issue, would be greatly appreciated, thank you!!

**Comment by FabianHoerst** [comment]
Thanks for the update! Indeed, I experienced the same behavior, but this seems reasonable to me as the domain shift between the pretraining task and segmentation is somehow large.

I will close the issue now. If you have any other questions, feel free to reach out again! 

Best, Fabian
